common/src/main/java/com/wynntils/screens/widgets/TextInputBoxWidget.java

```diff
@@ -40,34 +44,62 @@ public TextInputBoxWidget(
             int height,
             Component boxTitle,
             Consumer<String> onUpdateConsumer,
-            WynntilsSettingsScreen settingsScreen) {
+            SearchableScreen searchableScreen) {
         super(x, y, width, height, boxTitle);
         this.onUpdateConsumer = onUpdateConsumer;
-        this.settingsScreen = settingsScreen;
+        this.searchableScreen = searchableScreen;
     }
 
     public TextInputBoxWidget(
-            int x,
-            int y,
-            int width,
-            int height,
-            Consumer<String> onUpdateConsumer,
-            WynntilsSettingsScreen settingsScreen) {
+            int x, int y, int width, int height, Consumer<String> onUpdateConsumer, SearchableScreen searchableScreen) {
         super(x, y, width, height, TextComponent.EMPTY);
         this.onUpdateConsumer = onUpdateConsumer;
-        this.settingsScreen = settingsScreen;
+        this.searchableScreen = searchableScreen;
     }
 
     @Override
     public void render(PoseStack poseStack, int mouseX, int mouseY, float partialTick) {
+        checkForHeldBackspace();
+
         poseStack.pushPose();
 
         poseStack.translate(this.x, this.y, 0);
 
         this.renderBg(poseStack, McUtils.mc(), mouseX, mouseY);
 
+        String renderedText = getRenderedText(this.width - 4);
+
+        FontRenderer.getInstance()
+                .renderAlignedTextInBox(
+                        poseStack,
+                        renderedText,
+                        2,
+                        this.width,
+                        2,
+                        0,
+                        CommonColors.WHITE,
+                        HorizontalAlignment.Left,
+                        FontRenderer.TextShadow.NORMAL);
+
+        poseStack.popPose();
+    }
+
+    protected void checkForHeldBackspace() {
+        if (KeyboardUtils.isKeyDown(GLFW.GLFW_KEY_BACKSPACE)) {
```

Comment:
this is to get repeating backspace? Why are not all characters repeating? How do Vanilla do this? Is there some functionality they have that we could reuse? (You don't necessarily have to do that in this PR)

Reply:
I think legacy did `Keyboard.enableRepeatEvents()`. I cannot find that option atm, this should be good enough.

Reply:
I looked around the Vanilla code a bit now. There is a net.minecraft.client.gui.components.EditBox. It seems to be quite full-fledged. Why are we not using that?

Reply:
Ah. I see. I can try to use that for every text box. Another PR though?

Reply:
Another PR.